### PR TITLE
DASH-241. Refresh checkout page

### PR DESCRIPTION
**Pull request by andreyvEze**

### CODE REVIEW

File: client/src/api/orders.ts [hunk 1]
```diff
@@ -18,4 +18,10 @@ export default class Orders extends Base {
   ): Promise<OrdersUpdateResponse> {
     return this.apiClient.post(`orders/update/${id}`, { data });
   }
+  getNew(): Promise<OrdersCreateResponse> {
+    return this.apiClient.get(`orders/new`);
```
Comment (vmkarpenko) [inline client/src/api/orders.ts:22]: `orders/new` can be confusing with `orders/:id`. could we change this route to some special for this case

File: client/src/pages/CheckoutPage/CheckoutPageContext.tsx [hunk 2]
```diff
@@ -134,19 +129,23 @@ export const useContext = (): {
...
+  const [orderError, setOrderError] = useState<ApiError>(null);
 
   const {
     location: { state },
   } = history;
-  const { paymentProvider }: { paymentProvider?: PaymentProvider } =
-    state || {};
+  const {
+    orderParams: orderParamsFromLocation,
+  }: { orderParams?: CreateOrderActionData } = state || {}; // todo: implement setting order params from places where we going to checkout page
```
Comment (vmkarpenko) [inline client/src/pages/CheckoutPage/CheckoutPageContext.tsx:141]: Do we have task for this todo?

File: client/src/api/orders.ts (hunk 1 above)
Comment (andreyvEze) [reply client/src/api/orders.ts:22]: - `orders/active` 
- `orders/info/:id` or `orders/item/:id`

ok?

File: client/src/pages/CheckoutPage/CheckoutPageContext.tsx (hunk 2 above)
Comment (andreyvEze) [reply client/src/pages/CheckoutPage/CheckoutPageContext.tsx:141]: will create it if would not able to do this tomorrow

File: server/src/services/orders/New.mjs [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+const CART_TIMEOUT = 1000 * 60 * 30; // 30 min
+
+export default class OrdersList extends Base {
+  async execute() {
+    const order = await Order.findOne({
+      where: {
+        status: Order.STATUS.NEW,
+        userId: this.context.id,
+        createdAt: {
+          [Sequelize.Op.gt]: new Date(new Date().getTime() - CART_TIMEOUT),
```
[the remark this answers is not in the document]
Comment (andreyvEze) [reply server/src/services/orders/New.mjs:17]: before any new active order removed previous, but we were asked to keep them for history. let's re-discuss it

**Review by vmkarpenko — APPROVED** [review]

File: client/src/pages/CartPage/CartPageContext.tsx [hunk 4]
```diff
@@ -101,6 +105,10 @@ export const useContext = (): UseContextReturnType => {
   const [isPriceChanged, handlePriceChange] = useState(false);
   const [isUpdatingPrices, setIsUpdatingPrices] = useState(false);
   const [updatingPricesHasError, setUpdatingPricesHasError] = useState(false);
+  const [
+    checkoutApproving,
```
Comment (trukhilio) [inline client/src/pages/CartPage/CartPageContext.tsx:109]: Rename `checkoutApproving` to something else that represents better understanding of what is it.

File: client/src/utils.ts [hunk 5]
```diff
@@ -324,6 +361,25 @@ export const totalCost = (
   };
 };
 
+export const cartIsRelative = (
```
Comment (trukhilio) [inline client/src/utils.ts:364]: Could we move this util into separate util file named `cart`? In future we will move there other cart utils

**Pull request merged**